core/src/main/java/com/datastax/oss/driver/api/core/Cluster.java

```diff
@@ -116,6 +117,40 @@ default Metadata refreshSchema() {
     return CompletableFutures.getUninterruptibly(refreshSchemaAsync());
   }
 
+  /**
+   * Checks if all nodes in the cluster agree on a common schema version.
+   *
+   * <p>The goal of schema agreement is to ensure that a schema change has been propagated to all
+   * nodes before proceeding with the next queries (which might go to a different coordinator).
+   * Normally, you should not have to do that manually, because it's handled by the driver after all
+   * schema-altering queries. However, this method is provided in case schema agreement failed
+   * earlier and you want to check it again.
+   *
+   * <p>To check agreement, this method queries system tables to find out the schema version of all
+   * nodes that are currently {@code {@link NodeState#UP UP}}, and compares them. If the comparison
```

Comment:
To expound on that.  Just because the driver thinks a node is down, doesn't mean it's actually down (the driver could be actively ignoring it too).  Maybe it is best to ensure all nodes agree, not just the ones we can connect to.

Reply:
What if a node is really down? Schema agreement would have no chance of succeeding, as I imagine `system.peers` keeps the last known version.

Reply:
That's a good point, but we don't really have a way to know if a host is actually 'down' right?   This is probably too paranoid, but if you have some kind of network partition, and a schema change is made on one set of nodes, and also on another set, you could get a disagreement that way and i'm not sure how cassandra resolves these situations (it probably varies from release to release).   Since the 3.x driver already worked this way, I don't think we need to consider changing it here now (maybe some kind of future debate, i'll think on it some more).

It probably isn't a good idea to make schema changes if nodes are down/unavailable.  You could also argue that we should check for schema agreement before making a schema change (although it's not really possible on our end to know if a query is a schema change ahead of time).

Reply:
The goal of schema agreement is to protect *the client* from inconsistencies, e.g.:
* you create a table in one request and insert a row into it in the next, but the latter fails because it went to a different coordinator that hasn't sync'd yet,
* a new node comes up, you try to reprepare statements on it but it fails because it doesn't have all the tables,
* a schema change event comes in, you refresh but miss the update (admittedly harder because most of the time the refresh is done on the same control connection that received the event).

If the client thinks a node is down, it's not interacting with it, so none of these issues can happen. If the node is actually up and out of sync, it does not matter.